py/htm/algorithms/anomaly.py

```diff
@@ -17,30 +17,54 @@
 
 """
 
+There are currently several ways to get anomaly score:
+1) C++ Anomaly.hpp
+2) C++ & python TM.anomaly
+3) C++ & python AnomalyLikelihood
+4) this class
 
-There are two ways to use the code: using the
-:class:`.anomaly_li
+This class is pure python code calculating raw anomaly score.
+Was created for reason resulting from situation when TM.compute call is break down to individual steps.
 
-"""
+Simple calculates what is the overlap ratio between active columns
+and columns with predictive cells.
+
+anomaly 1.0 = all active columns were predicted (every active column is overlapping with column with predictive cell),
+anomaly 0.0 = none of active columns were predicted (none of active columns is overlapping with column with predictive cell)
 
+"""
+from htm.bindings.sdr import SDR
 
 class Anomaly:
+
+
   @staticmethod
   def calculateRawAnomaly(activeColSDR, predictiveCellsSDR ):
+    """
 
-    intersect = SDR(activeColSDR.dimensions)
-    intersect.intersection(activeColSDR, cellsToColumns(predictiveCellsSDR))
+    :param activeColSDR: SDR with active columns - one dimensional - like SDR(100)
+    :param predictiveCellsSDR: SDR with predictive cells - two dimensional - like SDR(100,30), means 30 cells per column
+    :return:
+    """
+    if len(activeColSDR.dimensions) !=1:
+      raise ValueError("activeColumns SDR must be of dimension 1")
+    if len(predictiveCellsSDR.dimensions) != 2:
+      raise ValueError("predictiveCells SDR must be of dimension 2")
```

Comment:
you can't do this check, as we can have multi-dimensional SP,TM. 
Please change this to the format as in `Anomaly.hpp`, that is: all inputs are in column format (check that they have same dims)